src/software/backend/simulation/physics/physics_robot.cpp

```diff
@@ -0,0 +1,120 @@
+#include "software/backend/simulation/physics/physics_robot.h"
+
+#include "shared/constants.h"
+#include "software/backend/simulation/physics/box2d_util.h"
+
+PhysicsRobot::PhysicsRobot(std::shared_ptr<b2World> world, const Robot& robot)
+    : robot_id(robot.id())
+{
+    // All the BodyDef must be defined before the body is created.
+    // Changes made after aren't reflected
+    createRobotPhysicsBody(world, robot);
+    setupRobotPhysicsBody(robot);
+    setupChickerPhysicsBody(robot);
+}
+
+PhysicsRobot::~PhysicsRobot()
+{
+    // Examples for removing bodies safely from
+    // https://www.iforce2d.net/b2dtut/removing-bodies
+    b2World* robot_body_world = robot_body->GetWorld();
+    if (bodyExistsInWorld(robot_body, robot_body_world))
+    {
+        robot_body_world->DestroyBody(robot_body);
+    }
+}
+
+void PhysicsRobot::createRobotPhysicsBody(std::shared_ptr<b2World> world,
+                                          const Robot& robot)
+{
+    robot_body_def.type = b2_dynamicBody;
+    robot_body_def.position.Set(robot.position().x(), robot.position().y());
+    robot_body_def.linearVelocity.Set(robot.position().x(), robot.position().y());
+    robot_body_def.angle           = robot.orientation().toRadians();
+    robot_body_def.angularVelocity = robot.angularVelocity().toRadians();
+
+    robot_body = world->CreateBody(&robot_body_def);
+}
+
+void PhysicsRobot::setupRobotPhysicsBody(const Robot& robot)
+{
+    // To create a polygon that approximates the shape of a robot, we find
+    // one of the points at the edge of the flat face at the front of the robot,
+    // and sweep around the back of the robot evenly distributing vertices until
+    // we reach the other edge of the flat face.
+
+    const unsigned int num_robot_body_vertices = b2_maxPolygonVertices;
+    b2Vec2 robot_body_vertices[num_robot_body_vertices];
+
+    Angle angle_to_edge_of_flat_face_relative_to_robot_orientation =
+        Vector(DIST_TO_FRONT_OF_ROBOT_METERS, FRONT_OF_ROBOT_WIDTH / 2.0).orientation();
+    Angle global_angle_to_edge_of_flat_face =
+        angle_to_edge_of_flat_face_relative_to_robot_orientation + robot.orientation();
+
+    Angle angle_to_sweep_across =
+        Angle::full() - (2 * angle_to_edge_of_flat_face_relative_to_robot_orientation);
+    Angle angle_increment = angle_to_sweep_across / (num_robot_body_vertices - 1);
+
+    for (unsigned int i = 0; i < num_robot_body_vertices; i++)
+    {
+        Angle angle_to_vertex = global_angle_to_edge_of_flat_face + (i * angle_increment);
+        // Shapes are defined relative to the body they are added to, so we do everything
+        // relative to the centre of the robot body, which is treated as (0, 0)
+        Vector vector_from_body_to_vertex =
+            Vector::createFromAngle(angle_to_vertex).normalize(ROBOT_MAX_RADIUS_METERS);
+        Point vertex           = Point(0, 0) + vector_from_body_to_vertex;
+        robot_body_vertices[i] = createVec2(vertex);
+    }
+
```

Comment:
Would look cleaner if we put this into a helper function such as `getRobotBodyVertices`

Reply:
Agreed. Refactored into new function.